lib/rust/ensogl/core/src/system/gpu/data/attribute.rs

```diff
@@ -204,9 +179,124 @@ impl {
             buffer.set_context(context);
         }
     }
+
+
+    // === Instance allocation ===
+
+    /// Allocate an element in the specified partition.
```

Comment:
Please expand these docs. Based on the current docs I do not really understand what it does and it looks like an important, complex thing.